core/protocolupgrade/engine.go

```diff
@@ -129,12 +130,17 @@ func (e *Engine) IsValidProposal(ctx context.Context, pk string, upgradeBlockHei
 		return errors.New("upgrade block earlier than current block height")
 	}
 
-	_, err := semver.Parse(TrimReleaseTag(vegaReleaseTag))
+	newv, err := semver.Parse(TrimReleaseTag(vegaReleaseTag))
 	if err != nil {
 		err = fmt.Errorf("invalid protocol version for upgrade received: version (%s), %w", vegaReleaseTag, err)
 		e.log.Error("", logging.Error(err))
 		return err
 	}
+
+	if semver.MustParse(e.currentVersion).GT(newv) {
```

Comment:
Should this not be GTE? because you can propose to upgrade to the current version to cancel your current proposal.